lib/scoped_search/query_builder.rb

```diff
@@ -485,6 +489,16 @@ def to_sql(builder, definition, &block)
             raise ScopedSearch::QueryNotSupported, "Don't know how to handle this operator node: #{operator.inspect} with #{children.inspect}!"
           end
         end
+
+        private
+
+        def validate_value(field, value)
+          validator = field.validator
+          if field.validator && field.only_explicit
+            valid = validator.call(value)
```

Comment:
It's a validator actually. I would expect it to return false when the validation fails.
I like the idea of `raise` in case of custom message - it simplifies the code. IMHO it's the best option - the validator returns either true or false (including "falsish" values like `nil`). The developer will be aware that nil fails the validation, since he will get the exception immediately (not like in case of `before_filter`, that the behavior is hidden from the developer ;) ).
If someone wants a custom message out of the validator - we will instruct him to `raise` an exception.

Sounds good?